x-pack/plugins/file_upload/public/components/geojson_upload_form/geojson_upload_form.tsx

```diff
@@ -42,35 +38,24 @@ interface State {
   hasFile: boolean;
   isPointsOnly: boolean;
   indexNames: string[];
+  updatedIndexName: string;
 }
 
 export class GeoJsonUploadForm extends Component<Props, State> {
-  private _isMounted = false;
-
   state: State = {
     hasFile: false,
     isPointsOnly: false,
     indexNames: [],
+    updatedIndexName: '',
   };
 
-  async componentDidMount() {
-    this._isMounted = true;
-    this._loadIndexNames();
-  }
-
-  componentWillUnmount() {
-    this._isMounted = false;
-  }
-
-  _loadIndexNames = async () => {
-    const indexNameList = await getExistingIndexNames();
-    const indexPatternList = await getExistingIndexPatternNames();
-    if (this._isMounted) {
-      this.setState({
-        indexNames: [...indexNameList, ...indexPatternList],
-      });
+  componentDidUpdate(prevProps: Props) {
```

Comment:
This can be removed once state is stored in a single location and always just pushed down

Reply:
sWe still need this as well. Updates to the name can come from two places: 1. the name of the file, 2. user input. For file name input, this updates the state here directly. For user input, the input is entered into `IndexNameForm` and via callback, `JSONUploadAndParse` state is updated and then passed down as props to `GeoJsonUploadForm` where it's grabbed and assigned to state here. Regardless of which input updates the name, this ensures it's always just a single name index name source passed to `IndexNameForm`.